GameOfRunes/GameOfRunes/GameEngine/Systems/TimerSystem.swift

```diff
@@ -18,23 +18,35 @@ class TimerSystem: GKComponentSystem<TimerComponent>, System {
     
     override func update(deltaTime seconds: TimeInterval) {
         for component in components {
-            updateComponent(component)
+            if component.isCountDown {
+                component.time -= seconds
+            } else {
+                component.time += seconds
+            }
+
+            updateComponent(component, seconds)
         }
     }
     
-    private func updateComponent(_ component: TimerComponent) {
-        guard CACurrentMediaTime() - component.lastUpdatedTime >= 1.0 else {
+    func updateComponent(_ component: TimerComponent, _ seconds: TimeInterval) {
+        guard let entity = component.entity as? Entity else {
             return
         }
-        
-        component.lastUpdatedTime = CACurrentMediaTime()
-        
-        if component.isCountdown {
-            component.currentTime = max(0, component.currentTime - 1)
-            // TODO: Check if is 0, then propagate call to game state machine for lose state.
-        } else {
-            component.currentTime += 1
-//            gameEngine?.spawnEnemy()
+        if component.time <= 0, var powerUpEntity = entity as? PowerUpEntity {
```

Comment:
For better code quality: use `entity.type` as a switch case, to run different methods for different entities. Also, add a todo to put fading into `PowerUpComponent`, entity itself should not have any variables